Plots/Meteograms/NCL_meteo_1.py

```diff
@@ -0,0 +1,291 @@
+"""
+NCL_meteo_1.py
+===============
+
+This script illustrates the following concepts:
+   - Drawing a meteogram
+   - Creating a color map using hex values
+   - Reversing the Y axis
+   - Explicitly setting tickmarks and labels on the bottom X axis
+   - Increasing the thickness of contour lines
+   - Drawing wind barbs
+   - Drawing a bar chart
+   - Changing the width and height of a plot
+   - Overlaying wind barbs and line contours on filled contours
+   - Changing the position of individual plots on a page
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/meteo_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/meteo_1_lg.png
+"""
+
+###############################################################################
+# Import necessary packages
+import xarray as xr
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.colors import ListedColormap, BoundaryNorm
+import cartopy.crs as ccrs
+
+import geocat.datafiles as gdf
+from geocat.viz import util as gvutil
+
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine
+# and load the data into xarray
+ds = xr.open_dataset(gdf.get("netcdf_files/meteo_data.nc"), decode_times=False)
+
+# Extract variables from the data
+tempisobar = ds.tempisobar
+levels = ds.levels
+taus = ds.taus
+rh = ds.rh
+ugrid = ds.ugrid
+vgrid = ds.vgrid
+rain03 = ds.rain03
+tempht = ds.tempht
+
+###############################################################################
+# Plot:
+
+# Generate figure (set its size (width, height) in inches)
+fig = plt.figure(figsize=(6, 10))
```

Comment:
Instead of using `plt.figure` you could to use `plt.subplots` and specify the ratio between the axes heights using the key words `height_ratios` and `gridspec_kw`. That way the pressure and wind barb map won't be squished vertically and can have an aspect ratio similar to the NCL version. This would also allow you to make three subplots stacked vertically with different heights instead of using inset axes. 

It's up to you if you want to use `plt.figure` or `plt.subplots`, but either way the top set of axes should be a bit taller.